app/models/financialDetails/WhatYouOweChargesList.scala

```diff
@@ -17,22 +17,43 @@
 package models.financialDetails
 
 import exceptions.MissingFieldException
-import models.outstandingCharges.OutstandingChargesModel
+import models.outstandingCharges.{OutstandingChargeModel, OutstandingChargesModel}
 import services.DateServiceInterface
+import views.html.partials.chargeSummary.ChargeSummaryCodingOut
 
 import java.time.LocalDate
 
 case class WhatYouOweChargesList(balanceDetails: BalanceDetails, chargesList: List[ChargeItem] = List(),
                                  outstandingChargesModel: Option[OutstandingChargesModel] = None,
                                  codedOutDocumentDetail: Option[ChargeItem] = None)(implicit val dateService: DateServiceInterface) {
 
-  private lazy val overdueChargeList: List[ChargeItem] = chargesList.filter(_.isOverdue())
+  lazy val overdueChargeList: List[ChargeItem] = chargesList.filter(x => x.isOverdue())
+
+  lazy val overdueOrAccruingInterestChargeList: List[ChargeItem] = chargesList.filter(x => x.isOverdue() || x.hasAccruingInterest)
+  lazy val overdueOutstandingCharges: List[OutstandingChargeModel] = outstandingChargesModel.toList.flatMap(_.outstandingCharges)
+    .filter(_.relevantDueDate.getOrElse(LocalDate.MAX).isBefore(dateService.getCurrentDate))
+
+  lazy val chargesDueWithin30DaysList: List[ChargeItem] = chargesList.filter(x => !x.isOverdue() && !x.hasAccruingInterest && isWithin30Days(x.dueDate))
+
+  lazy val chargesDueAfter30DaysList: List[ChargeItem] = chargesList.filter(x => !x.isOverdue() && !x.hasAccruingInterest && !isWithin30Days(x.dueDate))
+
+  def isWithin30Days(date: Option[LocalDate]): Boolean = {
```

Comment:
This is something we should move under DateServiceInterface so we can re-use this method;
I am quite sure we already have similar logic somewhere else